zkevm-circuits/src/evm_circuit/execution/error_oog_dynamic_memory.rs

```diff
@@ -0,0 +1,365 @@
+        let address = block.rws[step.rw_indices[rw_offset]].stack_value();
+        let size = block.rws[step.rw_indices[rw_offset + 1]].stack_value();
+        self.address
+            .assign(region, offset, Some(address.to_le_bytes()))?;
+        self.size.assign(region, offset, Some(size.to_le_bytes()))?;
+
+        let address_high = address_high::value::<F>(address.to_le_bytes());
+        assert_eq!(
+            address_high,
+            F::zero(),
+            "address overflow {} bytes",
+            N_BYTES_MEMORY_ADDRESS
+        );
+        self.address_in_range_high
+            .assign(region, offset, address_high)?;
+        let size_high = address_high::value::<F>(size.to_le_bytes());
+        assert_eq!(
+            size_high,
+            F::zero(),
+            "size overflow {} bytes",
+            N_BYTES_MEMORY_ADDRESS
+        );
+        self.size_in_range_high.assign(region, offset, size_high)?;
+
+        let address_value = address_low::value(address.to_le_bytes());
+        let size_value = address_low::value(size.to_le_bytes());
+        let expanded_address = address_value
+            .checked_add(size_value)
+            .expect("address overflow u64");
+        assert!(
+            expanded_address < (1u64 << (N_BYTES_MEMORY_ADDRESS * 8)),
+            "expanded address overflow {} bytes",
+            N_BYTES_MEMORY_ADDRESS
+        );
+        self.expanded_address_in_range.assign(
+            region,
+            offset,
+            F::from(expanded_address),
+            F::from(1u64 << (N_BYTES_MEMORY_ADDRESS * 8)),
+        )?;
+
+        self.memory_expansion.assign(
+            region,
+            offset,
+            step.memory_word_size(),
+            [expanded_address],
+        )?;
+
+        self.insufficient_gas.assign(
+            region,
+            offset,
+            F::from(step.gas_left),
+            F::from(step.gas_cost),
+        )?;
+
+        self.rw_counter_end_of_reversion.assign(
+            region,
+            offset,
+            Value::known(F::from(call.rw_counter_end_of_reversion as u64)),
+        )?;
+
+        self.restore_context.assign(
+            region,
+            offset,
+            block,
+            call,
+            step,
+            if opcode == OpcodeId::CREATE { 5 } else { 4 },
+        )?;
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::test_util::CircuitTestBuilder;
+    use eth_types::{bytecode, word, Bytecode, ToWord};
+    use mock::test_ctx::helpers::account_0_code_account_1_no_code;
+    use mock::{eth, TestContext, MOCK_ACCOUNTS};
+
+    #[test]
+    fn test() {
+        let codes = [
+            bytecode! {
+                PUSH8(0xFF)
+                PUSH32(word!("0xffffffff")) // offset
+                RETURN
+                STOP
+            },
+            bytecode! {
+                PUSH8(0xFF)
+                PUSH32(word!("0xffffffff")) // offset
+                REVERT
+                STOP
```

Comment:
this STOP will not execute as REVERT will terminate current call context I think. suggest to remove .

Reply:
There are lots of this kind of bytecode in other files. Is this kind of a code style?

Reply:
should be not found previously 😄